src/middlewared/middlewared/plugins/directoryservices.py

```diff
@@ -215,8 +215,24 @@ async def get_last_password_change(self, domain=None):
 
         return {"dbconfig": stored_ts, "secrets": passwd_ts}
 
+    async def __kerberos_start_retry(self, retries=10):
+        while retries:
```

Comment:
On the infinitesimal chance `retries` skips `0` and goes negative.  Maybe `while retries >= 0`